app/services/hyrax/messenger_service.rb

```diff
@@ -0,0 +1,8 @@
+module Hyrax
+  class MessengerService
+    def self.deliver(sender, recipients, body, subject, *args)
+      sender.send_message(recipients, body, subject, *args)
+      StreamNotificationsJob.perform_later(recipients)
```

Comment:
The MessengerService is used to send all notifications. After every message is sent (possibly to multiple recipients), queue a new StreamNotificationsJob.